elastio-s3-changelog/cloudformation-single.yaml

```diff
@@ -0,0 +1,263 @@
+      Tags:
+        'elastio:resource': 'true'
+
+  queueDepthAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Condition: EnableQrts
+    Properties:
+      AlarmDescription: Alarm if queue depth increases too much.
+      ActionsEnabled: false
+      Namespace: AWS/SQS
+      MetricName: ApproximateNumberOfMessagesVisible
+      Dimensions:
+        - Name: QueueName
+          Value: !GetAtt queue.QueueName
+      Statistic: Maximum
+      Period: 60
+      EvaluationPeriods: 1
+      Threshold: 5
+      ComparisonOperator: GreaterThanThreshold
+
+  queueDepthAlarmSubscription:
+    Type: AWS::Events::Rule
+    Condition: EnableQrts
+    Properties:
+      Description: Track queue depth alarm state
+      EventPattern:
+        source: [aws.cloudwatch]
+        detail-type: [CloudWatch Alarm State Change]
+        resources:
+          - !GetAtt queueDepthAlarm.Arn
+        detail:
+          state:
+            value: [ALARM]
+      Targets:
+        - Arn: !Sub 'arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:elastio-scheduler-handler-s3-changelog-qrts'
+          Id: event-handler
+          Input: !Sub >
+            {
+              "bucket_name": "${BucketName}",
+              "queue_url": "${queue}",
+              "queue_name": "${queue.QueueName}"
+            }
+
+  queueDepthAlarmPermission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      FunctionName: !Sub 'arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:elastio-scheduler-handler-s3-changelog-qrts'
+      Action: lambda:InvokeFunction
+      Principal: events.amazonaws.com
+      SourceArn: !GetAtt queueDepthAlarmSubscription.Arn
+
+  queueOldestMessageAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Condition: EnableQrts
+    Properties:
+      AlarmDescription: Alarm if the age of the oldest massage is too big.
+      ActionsEnabled: false
+      Namespace: AWS/SQS
+      MetricName: ApproximateAgeOfOldestMessage
+      Dimensions:
+        - Name: QueueName
+          Value: !GetAtt queue.QueueName
+      Statistic: Maximum
+      Period: 60
+      EvaluationPeriods: 1
+      Threshold: 600 # 10 minutes
+      ComparisonOperator: GreaterThanThreshold
+
+  queueOldestMessageAlarmSubscription:
+    Type: AWS::Events::Rule
+    Condition: EnableQrts
+    Properties:
+      Description: Track queue oldest message alarm state
+      EventPattern:
+        source: [aws.cloudwatch]
+        detail-type: [CloudWatch Alarm State Change]
+        resources:
+          - !GetAtt queueOldestMessageAlarm.Arn
+        detail:
+          state:
+            value: [ALARM]
+      Targets:
+        - Arn: !Sub 'arn:aws:lambda:${AWS::Region}:${AWS::AccountId}:function:elastio-scheduler-handler-s3-changelog-qrts'
+          Id: event-handler
+          Input: !Sub >
+            {
+              "bucket_name": "${BucketName}",
+              "queue_url": "${queue}",
+              "queue_name": "${queue.QueueName}"
+            }
+
+  queueOldestMessageAlarmPermission:
```

Comment:
I think we should define the resource-based policy for lambda in our region-level stack instead of here. We just allow invocation from any eventbridge rule for our lambdas.